pkg/serviceproxy/service_proxy_transport_test.go

```
@@ -0,0 +1,170 @@
+	if transport.ForceAttemptHTTP2 {
+		t.Fatal("ForceAttemptHTTP2 must remain disabled for SPDY upgrades")
+	}
+}
+
+func TestServeHTTPReusesSharedTransport(t *testing.T) {
+	transport := &recordingRoundTripper{}
+	s := &serviceProxy{proxyTransport: transport}
+
+	sendRequest := func(t *testing.T) {
+		t.Helper()
+		req := httptest.NewRequest(http.MethodGet, "https://service-proxy.example/healthz", nil)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)